cfme/tests/containers/test_smart_management.py

```diff
@@ -1,77 +1,72 @@
-from cfme.containers.pod import Pod, PodCollection
-from cfme.containers.template import Template, TemplateCollection
-from cfme.containers.container import Container, ContainerCollection
-from cfme.utils.log import create_sublogger
+from cfme.containers.provider import ContainersProvider, ContainersTestItem
+from cfme.containers.project import Project
+from cfme.containers.image_registry import ImageRegistry
+from cfme.containers.node import Node
+from cfme.containers.pod import Pod
+from cfme.containers.template import Template
+from cfme.containers.container import Container
 from cfme.utils.wait import wait_for
-from cfme.utils.appliance.implementations.ui import navigate_to
+
 from cfme.utils.blockers import BZ
 
 pytestmark = [
     pytest.mark.usefixtures('setup_provider'),
     pytest.mark.tier(1),
     pytest.mark.provider([ContainersProvider], scope='function')
 ]
-logger = create_sublogger("smart_management")
 
 TEST_ITEMS = [
     pytest.mark.polarion('CMP-9948')(ContainersTestItem(
-        Container, 'CMP-9948', collection_obj=ContainerCollection)),
+        Container, 'CMP-9948', collection_obj="containers", keyword_name="name")),
     pytest.mark.polarion('CMP-10320')(ContainersTestItem(
-        Template, 'CMP-10320', collection_obj=TemplateCollection)),
+        Template, 'CMP-10320', collection_obj="container_templates", keyword_name="name")),
     pytest.mark.polarion('CMP-9992')(ContainersTestItem(
-        ImageRegistry, 'CMP-9992', collection_obj=ImageRegistryCollection)),
-    pytest.mark.polarion('CMP-9981')(ContainersTestItem(
-        Image, 'CMP-9981', collection_obj=ImageCollection)),
+        ImageRegistry, 'CMP-9992', collection_obj="container_image_registries",
+        keyword_name="host")),
     pytest.mark.polarion('CMP-9964')(ContainersTestItem(
-        Node, 'CMP-9964', collection_obj=NodeCollection)),
+        Node, 'CMP-9964', collection_obj="container_nodes", keyword_name="name")),
     pytest.mark.polarion('CMP-9932')(ContainersTestItem(
-        Pod, 'CMP-9932', collection_obj=PodCollection)),
+        Pod, 'CMP-9932', collection_obj="container_pods", keyword_name="name")),
     pytest.mark.polarion('CMP-9870')(ContainersTestItem(
-        Project, 'CMP-9870', collection_obj=ProjectCollection)),
+        Project, 'CMP-9870', collection_obj="container_projects", keyword_name="name")),
     pytest.mark.polarion('CMP-9854')(ContainersTestItem(
-        ContainersProvider, 'CMP-9854', collection_obj=None))
+        ContainersProvider, 'CMP-9854', collection_obj="containers_providers",
+        keyword_name="name"))
 ]
 
 
-def get_object_name(obj):
-    return obj.__module__.title().split(".")[-1]
+@pytest.fixture
+def get_entity(collection):
+
+    # Map all object by name
+    mapping = {item.name: item for item in collection.all()}
 
+    # Filter only the unique items
+    unique_items = [key for key in set(mapping.keys()) if mapping.keys().count(key) == 1]
```

Comment:
OPTIONAL: .keys() is default for dict. so, you don't need to specify it
REQUIRED: you use dict. dict cannot have two elements with the same key. Please check your logic.